### PR TITLE
PR: Send comm config on every message to avoid TimeoutError's

### PR DESCRIPTION
**Pull request by impact27**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

File: spyder/plugins/editor/widgets/codeeditor.py [hunk 1]
```diff
@@ -1558,7 +1558,8 @@ def update_debugger_panel_state(self, state, last_step, force=False):
             return
         elif state and 'fname' in last_step:
             fname = last_step['fname']
-            if osp.normcase(fname) == osp.normcase(self.filename):
+            if (fname and self.filename
+                    and osp.normcase(fname) == osp.normcase(self.filename)):
```
Comment (ccordoba12) [inline spyder/plugins/editor/widgets/codeeditor.py:1562]: Why is this change necessary?

File: spyder/plugins/ipythonconsole/comms/kernelcomm.py [hunk 2]
```diff
@@ -120,6 +122,12 @@ def remote_call(self, interrupt=False, blocking=False, callback=None,
             comm_id=comm_id, timeout=timeout)
 
     # ---- Private -----
+    def on_incoming_call(self, call_dict):
+        """A call was recieved"""
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole/comms/kernelcomm.py:126]: recieved -> received

File: spyder/plugins/ipythonconsole/comms/kernelcomm.py [hunk 3]
```diff
@@ -120,6 +122,12 @@ def remote_call(self, interrupt=False, blocking=False, callback=None,
...
+        if "Comm port" in call_dict:
+            self._set_comm_port(call_dict["Comm port"])
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole/comms/kernelcomm.py:128]: `Comm port` -> `comm_port`.

File: spyder/plugins/editor/widgets/codeeditor.py (hunk 1 above)
Comment (impact27) [reply spyder/plugins/editor/widgets/codeeditor.py:1562]: Apparently one of the two can be `None` if the debugging starts too quickly and `osp.normcase` doesn't like `None`

**Comment by impact27** [comment]
I am surprised by these failures. `spyder/plugins/ipythonconsole/tests/test_ipythonconsole.py::test_save_history_dbg` shouldn't involve comms at all because the pdb history is handeld in the fontend side.

**Comment by ccordoba12** [comment]
Tests are still failing. Should I revert spyder-ide/spyder-kernels#231 and leave this for 4.2.0? I really don't want potentially disruptive, last minute changes to be merged.

**Comment by impact27** [comment]
> Tests are still failing. Should I revert [spyder-ide/spyder-kernels#231](https://github.com/spyder-ide/spyder-kernels/pull/231) and leave this for 4.2.0? I really don't want potentially disruptive, last minute changes to be merged.

The tests are unrelated to this PR: https://github.com/spyder-ide/spyder/pull/13201

**Comment by ccordoba12** [comment]
Thanks for thinking about that @impact27! I'm going to close that PR to fix our tests in another one.

File: spyder/plugins/editor/widgets/codeeditor.py (hunk 1 above)
Comment (ccordoba12) [reply spyder/plugins/editor/widgets/codeeditor.py:1562]: Thanks for the explanation!

**Comment by ccordoba12** [comment]
@impact27, I fixed our tests in PR #13204 (it was due to an issue with the latest `ipykernel` version, released yesterday). Please merge with `4.x` to get the fix, so I can merge this one.

File: external-deps/spyder-kernels/.gitrepo [hunk 4]
```diff
@@ -7,6 +7,6 @@
 	remote = https://github.com/spyder-ide/spyder-kernels.git
 	branch = 1.x
 	commit = 1a9d44950edb462188f52cec51c975b80f5f0f6f
-	parent = adbc02315db0e49bbaa2cd6dae94526264398fc1
+	parent = ff7e4d25db6a366b51521870d76274a5e63dd0e8
```
Comment (ccordoba12) [inline external-deps/spyder-kernels/.gitrepo:10]: Please revert this change. It's not necessary because the latest version in the spyder-kernels subrepo contains the changes in spyder-ide/spyder-kernels#231.
